.github/workflows/ci.yml

```diff
@@ -0,0 +1,46 @@
+name: Build
+on:
+  push:
+    branches:
+      - master
+  pull_request: {}
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macOS-latest, windows-latest]
+        ghc: ['8.0', '8.2', '8.4', '8.6', '8.8', '8.10']
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-haskell@v1.1.3
+      id: setup-haskell-cabal
+      with:
+        ghc-version: ${{ matrix.ghc }}
+        cabal-version: '3.2'
+    - name: Update cabal package database
+      run: |
+        cabal update
+        cabal freeze
+    - uses: actions/cache@v2
+      name: Cache cabal stuff
+      with:
+        path: |
+          ${{ steps.setup-haskell-cabal.outputs.cabal-store }}
+          dist-newstyle
+          tests/dist-newstyle
+        key: ${{ runner.os }}-${{ matrix.ghc }}-${{ hashFiles('cabal.project.freeze') }}
+        restore-keys: |
+          ${{ runner.os }}-${{ matrix.ghc }}-
+    - name: Build
+      run: |
+        cabal build all
+    - name: Test
+      run: |
+        cd tests && cabal test
+    - name: SDist
+      run: |
+        cabal sdist all
```

Comment:
How about including haddocks, and particularly benchmarks? See https://github.com/haskell/bytestring/blob/1f97c4c9707e02b1f347b3ec99da851d0807a25f/.travis.yml#L51-L54

Reply:
Sure, added haddocks and benchmarks. They are visible in this run: https://github.com/ethercrow/bytestring/actions/runs/336677273